src/logic/cmp.rs

```diff
@@ -130,33 +154,24 @@ impl ApInt {
             ))
     }
 
-    /// Signed less-than (`slt`) comparison between `self` and `rhs`.
-    /// 
-    /// # Note
-    /// 
-    /// - `checked_` for this function means that it checks the bit widths
-    /// - Returns `Ok(true)` if `self < rhs`.
-    /// - Interprets both `ApInt` instances as **signed** values.
-    /// 
-    /// # Errors
-    /// 
-    /// - If `self` and `rhs` have unmatching bit widths.
+    /// Signed less-than (`slt`) comparison between `self` and `rhs`, meaning the returned boolean
+    /// indicates if `self < rhs` for the **signed** interpretation of `ApInt`s.
     pub fn checked_slt(&self, rhs: &ApInt) -> Result<bool> {
         let lhs = self;
         lhs.zip_access_data(rhs).and_then(|zipped| {
             match zipped {
                 ZipDataAccess::Inl(lhs, rhs) => {
-                    let infate_abs = digit::BITS - self.width().to_usize();
+                    let infate_abs = Digit::BITS - self.width().to_usize();
                     let lhs = (lhs.repr() << infate_abs) as i64;
                     let rhs = (rhs.repr() << infate_abs) as i64;
                     Ok(lhs < rhs)
                 }
                 ZipDataAccess::Ext(_, _) => {
                     match (lhs.sign_bit(), rhs.sign_bit()) {
-                        (Bit::Unset, Bit::Unset) => lhs.checked_ult(rhs),
-                        (Bit::Unset, Bit::Set  ) => Ok(false),
-                        (Bit::Set  , Bit::Unset) => Ok(true),
-                        (Bit::Set  , Bit::Set  ) => rhs.checked_ugt(lhs)
+                        (false, false) => lhs.checked_ult(rhs),
```

Comment:
I know I already stated this but I really prefer the `Bit::Zero` case. Maybe we should add something like
```rust
match (u8::from(lhs.sign_bit()), u8::from(rhs.sign_bit())) {
    (0, 0) => ..,
    (0, 1) => ..,
    (1, 0) => ..,
    (1, 1) => ..,
}
```